openmdao/core/component.py

```diff
@@ -373,6 +388,29 @@ def setup_partials(self):
         """
         pass
 
+    def _run_root_only(self):
+        """
+        Return the value of the run_root_only option and check for possible errors.
+
+        Returns
+        -------
+        bool
+            True if run_root_only is active.
+        """
+        if self.options['run_root_only']:
+            if self.comm.size > 1 or (self._full_comm is not None and self._full_comm.size > 1):
+                if self._has_distrib_vars:
```

Comment:
Can't all of these checks be done during final setup, instead of during every api call on the component?

Reply:
My reason for doing it this way was that there isn't an existing setup function where these checks seemed to fit, and I didn't really want to create another function.  I could have crammed the checks into some existing setup function, but it didn't seem worth it, considering that these checks are not expensive.